src/core/exec-invoke.c

```diff
@@ -5724,10 +5809,27 @@ int exec_invoke(
                         command,
                         needs_sandboxing,
                         have_cap_sys_admin,
+                        bpffs_socket_fd,
                         exit_status);
         if (r < 0)
                 return r;
 
+        if (context->private_bpf != PRIVATE_BPF_NO) {
+                r = pidref_wait_for_terminate_and_check("(sd-bpffs)", &bpffs_pidref, /* flags = */ 0);
+                if (r < 0) {
+                        *exit_status = EXIT_BPF;
+                        return r;
+                }
+                /* If something strange happened with the child, let's consider this fatal, too */
+                if (r != EXIT_SUCCESS) {
+                        *exit_status = EXIT_BPF;
+                        if (read(bpffs_errno_pipe, &r, sizeof(r)) == sizeof(r))
+                                return log_debug_errno(r, "bpffs helper exited with error: %m");
+                        return log_debug_errno(errno, "Failed to read from the bpffs helper errno pipe: %m");
```

Comment:
read() might return short, who knows, and you'll output errno here, which is not initialized in case of a short read